plugins/qiskit_executor/schemas.py

```diff
@@ -12,25 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import Optional
 import marshmallow as ma
+from celery.utils.log import get_task_logger
 from qhana_plugin_runner.api.util import (
     FrontendFormBaseSchema,
     FileUrl,
 )
-from celery.utils.log import get_task_logger
+from qhana_plugin_runner.util.logging import redact_log_data
 
 
 TASK_LOGGER = get_task_logger(__name__)
 
 
-class CircuitSelectionInputParameters:
+class CircuitParameters:
     def __init__(
         self,
         circuit: str,
-        executionOptions: str,
-        shots: int,
-        ibmqToken: str,
-        backend: str,
+        executionOptions: Optional[str],
+        shots: Optional[int],
```

Comment:
If they allow none, then you should use `if params.shots` instead of `if params.shots != ""` to test for empty/mising values. (This will currently not produce any bugs in `routes.py`, but may lead to some surprises if the file gets refactored in the future.) VSCode and pycharm should both have a feature to quickly find references/usages of these parameters